packages/fhir-router/src/graphql/graphql.ts

```diff
@@ -117,6 +117,8 @@ export async function graphqlHandler(
 
   if (includesMutations(query)) {
     repo.setMode(RepositoryMode.WRITER);
+  } else {
+    repo.setMode(RepositoryMode.READER);
```

Comment:
🤔 

I'm trying to remember why we used the original implementation...

I think the idea was "if we're already in `writer` mode, then we need to stay in `writer` mode"

In practice, that's probably not something that happens (?)  Maybe if someone tries to run a GraphQL request inside of a batch request?  I'm not sure if we actually even support that?